api/python/provisioner/commands/validator/validator.py

```diff
@@ -0,0 +1,250 @@
+import logging
+from abc import abstractmethod
+from pathlib import Path
+from typing import Dict, Callable
+
+from ...errors import ValidationError
+
+logger = logging.getLogger(__name__)
+
+
+class SchemeValidator:
+
+    """Base Scheme Validator. It defines the interface for child classes."""
+
+    def __init__(self):
+        pass
+
+    @abstractmethod
+    def validate(self, *args, **kwargs):
+        """
+        Abstract Scheme validation method.
+
+        Parameters
+        ----------
+        args
+            list of positional arguments
+        kwargs
+            keyword-only arguments
+
+        Returns
+        -------
+        None
+
+        Raises
+        ------
+        ValidationError
+            If validation is failed.
+        """
+        pass
+
+
+class Validator:
```

Comment:
How it differs from the `SchemeValidator` class?

Reply:
I planned to distinguish these two classes. `SchemeValidator` is a complex class-validator which used only for different scheme validations. For this moment I left only `Validator` base-class.